irods/test/connection_test.py

```diff
@@ -34,6 +34,29 @@ def test_connection_destructor(self):
         self.assertTrue(conn._disconnected)
         conn.release(destroy=True)
 
+    def test_server_version_without_authentication__issue_688(self):
+        sess = self.sess
+
+        # Make a session object that cannot authenticate.
+        non_authenticating_session = irods.session.iRODSSession(
+            host=sess.host,
+            port=sess.port,
+            user=sess.username,
+            zone=sess.zone,
+            # No password.
+        )
+
+        # Test server_version_without_auth method returns a value.
+        version_tup = non_authenticating_session.server_version_without_auth()
+
+        # Test returned value is non-empty "version" tuple, i.e. holds only integer values.
+        self.assertTrue(len(version_tup) > 0)
```

Comment:
Could be `assertGreater`. If you're into it.

Reply:
Done!